Makefile

```diff
@@ -0,0 +1,79 @@
+ARGS = $(filter-out $@,$(MAKECMDGOALS))
+MAKEFLAGS += --silent
+
+list:
+	sh -c "echo; $(MAKE) -p no_targets__ | awk -F':' '/^[a-zA-Z0-9][^\$$#\/\\t=]*:([^=]|$$)/ {split(\$$1,A,/ /);for(i in A)print A[i]}' | grep -v '__\$$' | grep -v 'Makefile'| sort"
```

Comment:
Suggestion, two: `grep -v '__\$$' | grep -v 'Makefile'` can be combined into one like:

    grep -v -e '__\$$' -e 'Makefile'